### ISSUE TITLE
Feature Request:MNASNET on tf-slim

### ISSUE THREAD

**Issue by rootkitchao** [tensorflow/models#5761]
### System information
- **What is the top-level directory of the model you are using**:tensorflow/models/
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**:Yes
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**:MS Windows 10 Pro 64bit Build 17763
- **TensorFlow installed from (source or binary)**:binary（tensorflow-gpu)
- **TensorFlow version (use command below)**:1.10.0-rc1-19-g656e7a2b34' 1.10.0
- **Bazel version (if compiling from source)**:N/A
- **CUDA/cuDNN version**:9.0/7.4
- **GPU model and memory**:NVIDIA Geforce RTX2080TI 11GB
- **Exact command to reproduce**:N/A

### Source code / logs
N/A

Dear tensorflower:
I have read the paper on MNASNET(https://arxiv.org/abs/1807.11626), but I have not found the code.I am planning to write a custom code for MNASNET.But before that, I want to ask if there is any plan to add MNASNET to TF-Slim.Thanks.

**Comment by rootkitchao** [comment]
I have implemented the standard MNASNET myself. At present, the 1.1M step is trained on the Imagenet-1K training set with 2*NVIDIA RTX 2080TI to achieve 68% accuracy. The rest of the training will take some time. But i still have no idea about the architectural details of MNASNET-65 and MNASNET-92.

**Comment by holyhao** [comment]
Mnasnet model and config  for object detection API will be highly appreciated as well.

**Comment by seefun** [comment]
Mnasnet model and config for object detection API will be highly appreciated +1

**Comment by saberkun** [comment]
We have an implementation https://github.com/tensorflow/tpu/tree/master/models/experimental/mnasnet
as a reference model + an updated model architecture. We will release pre-trained checkpoints/savedmodels/tflite very soon.

**Comment by rootkitchao** [comment]
> We have an implementation https://github.com/tensorflow/tpu/tree/master/models/experimental/mnasnet
> as a reference model + an updated model architecture. We will release pre-trained checkpoints/savedmodels/tflite very soon.
Thank you very much for your work.

**Comment by rootkitchao** [comment]
I used the wrong hyperparameter in the previous training (similar to Mobilenet V2).After training 3.0M steps with two NVIDIA Geforce RTX2080TIs, I only got 72.6 correct rate.Now I am trying to train using the hyperparameters in the paper.

**Comment by zychen2016** [comment]
But Object Detection pre-trained model?

**Comment by rootkitchao** [comment]
After modifying the hyperparameter.I trained 3,764,000 steps with two NVIDIA Geforce RTX2080TI graphics cards.The model achieved an accuracy of 73.84% on ILSVRC2012.The accuracy rate is now very close to the paper.
I will release the pre-training model and source code as soon as possible.
![image](https://user-images.githubusercontent.com/40640909/53271509-a866b180-3729-11e9-8d2d-ccfb48103a35.png)

**Comment by saberkun** [comment]
Thanks for the update. I feel it takes more steps to get higher accuracy as the learning rate decay overtime.
Beyond the model (mnasnet-b1) described in the paper, the author released an newer version which has better accuracy.  https://github.com/tensorflow/tpu/blob/master/models/official/mnasnet/mnasnet_models.py (mnasnet-a1)

**Comment by rootkitchao** [comment]
> Thanks for the update. I feel it takes more steps to get higher accuracy as the learning rate decay overtime.
> Beyond the model (mnasnet-b1) described in the paper, the author released an newer version which has better accuracy.  https://github.com/tensorflow/tpu/blob/master/models/official/mnasnet/mnasnet_models.py (mnasnet-a1)
Yes, I am continuing to train the model.And thank you very much for telling me about mnasnet-a1.

**Comment by rootkitchao** [comment]
![image](https://user-images.githubusercontent.com/40640909/53474714-ce34e300-3aa8-11e9-824c-449fc9063745.png)
After training 3944238 steps, the accuracy reached 74.0%.Training is still going on.But it seems that the accuracy can only reach 74.0%.At present, 4.182M steps have been trained, but the accuracy has not been further improved, only recall5 has reached 92%.
![image](https://user-images.githubusercontent.com/40640909/53474840-384d8800-3aa9-11e9-8c03-0eb2e7572908.png)

**Comment by holyhao** [comment]
@rootkitchao @saberkun  Hi, will you release the pre-trained model?

**Comment by rootkitchao** [comment]
> @rootkitchao @saberkun  Hi, will you release the pre-trained model?

I am sorry to answer this late, the pre-trained model has been released, and other parts have some minor problems to solve.
https://github.com/rootkitchao/mnasnet_tensorflow

**Comment by saeed68gm** [comment]
@rootkitchao Thanks a lot for your work. Do you plan on releasing the mnasnet-a1 pretrained weights as well?

**Comment by rootkitchao** [comment]
> @rootkitchao Thanks a lot for your work. Do you plan on releasing the mnasnet-a1 pretrained weights as well?
I have now ported tpu/models/official/mnasnet to the GPU. Unfortunately, there are some bugs that need to be solved.
1.There is a problem with distribute.Strategy when using ExponentialMovingAverage(https://github.com/tensorflow/tensorflow/issues/27392).This bug has been reported and has not been fixed yet.
2.Mixed training performance is very low when using deep separation convolution(https://github.com/tensorflow/tensorflow/issues/27780).This bug needs to be checked further.
I have bought some new graphics cards and will start training mnasnet a1 once the bugs are fixed.

**Comment by saeed68gm** [comment]
@rootkitchao Thank you! Keep me posted

**Comment by tensorflowbutler** [comment]
Hi There,
 We are checking to see if you still need help on this, as this seems to be considerably old issue. Please update this issue with the latest information, code snippet to reproduce your issue and error you are seeing.
 If we don't hear from you in the next 7 days, this issue will be closed automatically. If you don't need help on this issue any more, please consider closing this.
